# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
# Commit: 06ae2b9a2703
# Full Hash: 06ae2b9a2703e5b184ef8d41be63aff989eb0ee8
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Don't access GetStats() in AudioConduit::GetRemoteSSRC(). r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D103492
# ==============================================================================

diff -r 88a9fe04aaa5 -r 06ae2b9a2703 dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Mon Jan 18 15:28:29 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Fri Jan 29 15:51:59 2021 +0100
@@ -115,16 +115,11 @@
 }
 
 bool WebrtcAudioConduit::GetRemoteSSRC(uint32_t* ssrc) {
-  {
-    MutexAutoLock lock(mMutex);
-    if (!mRecvStream) {
-      return false;
-    }
-
-    const webrtc::AudioReceiveStream::Stats& stats = mRecvStream->GetStats();
-    *ssrc = stats.remote_ssrc;
+  MOZ_ASSERT(NS_IsMainThread());
+  if (!mRecvStream) {
+    return false;
   }
-
+  *ssrc = mRecvStreamConfig.rtp.remote_ssrc;
   return true;
 }
 
